users/tests/test_user.py

```diff
@@ -72,3 +74,35 @@ def test_no_authenticated_client(client, django_user_model):
     response = client.get('/profile/')
     assert response.status_code == 302
     assert response.url == '/login/?next=/profile/'
+
+
+def test_update_user(client, django_user_model):
+    username = "Test-user"
+    password = "123456"
+    django_user_model.objects.create_user(username=username, password=password)
+    client.username = 'updated-username'
+    assert client.username == 'updated-username'
+    client.email = 'update@mail.com'
+    assert client.email == 'update@mail.com'
+
+
+class TestForms(TestCase):
+    def test_UserRegistrationForm(self):
+        form_data = {
+            'username': 'username',
+            'email': 'Test@example.com',
+            'password1': 'test-password',
+            'password2': 'test-password',
+        }
+        form = UserRegistrationForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+    def test_UserUpdateForm(self):
```

Comment:
I think you should add a test that checks if the update worked (for both - UserUpdate and ProfileUpdate). 
maybe try to save the form and check the fields did update.

Reply:
+2 on that, see comment above related to forms testing

Reply:
@Yarboa I added tests for invalid forms and checked that we can't save them.